DatadogSessionReplay/Sources/Writers/ResourcesWriter.swift

```diff
@@ -16,25 +16,99 @@ internal protocol ResourcesWriting {
 
 internal class ResourcesWriter: ResourcesWriting {
     /// An instance of SDK core the SR feature is registered to.
-    private weak var core: DatadogCoreProtocol?
+    private let scope: FeatureScope?
+    private let telemetry: Telemetry
+
+    @ReadWriteLock
+    private var knownIdentifiers = Set<String>() {
+        didSet {
+            if let knownIdentifiers = knownIdentifiers.asData() {
```

Comment:
resources come in bulks per view tree capture. Because of that it felt good to update the data store on `didSet`.